### PR TITLE
core: close call when exceptions occur in onReady

### PR DESCRIPTION
**Pull request by carl-mastrangelo**
This is different than #1652

### CODE REVIEW

File: core/src/main/java/io/grpc/internal/ClientCallImpl.java [hunk 1]
```diff
@@ -466,6 +466,12 @@ public final void runInContext() {
             closed = true;
             cancelListenersShouldBeRemoved = true;
             observer.onClose(savedStatus, savedTrailers);
+          } catch (RuntimeException e) {
```
Comment (ejona86) [inline core/src/main/java/io/grpc/internal/ClientCallImpl.java]: Are these catches so you could put the comment some place?

**Comment by ejona86** [comment]
@carl-mastrangelo LGTM

File: core/src/main/java/io/grpc/internal/ClientCallImpl.java (hunk 1 above)
Comment (carl-mastrangelo) [inline core/src/main/java/io/grpc/internal/ClientCallImpl.java]: yes, I had intended to fill them with log statements, but didn't know if they should be caught by the executor

Comment (carl-mastrangelo) [inline core/src/main/java/io/grpc/internal/ClientCallImpl.java]: Actually, this seems mostly the same as #1979, so I am going to drop it.

**Pull request closed without merging**